### ISSUE TITLE
4D5307ED - Banjo Kazooie: Nuts & Bolts

### ISSUE THREAD

**Issue by DrChat** [xenia-project/game-compatibility#49]
Tested with: benvanik/xenia@08acb6b
[Marketplace](http://marketplace.xbox.com/en-US/Product/Banjo-Kazooie-N-n-B/66acd000-77fe-1000-9115-d8024d5307ed)

Deadlocks early while loading files @ 0x8226CAF8. Address containing the event handle is never read from until this call at runtime. Not exactly sure if this handle is directly related to I/O operations.

If I force it past this wait (by making the kernel call fail on this specific handle) the game appears to run fine for a while, then it deadlocks waiting on the main audio thread (spawned by the game).

**Comment by krzysztof860** [comment]
Also state-intro, but logos are displayed correctly. After displaying second logo Xenia crashes and memory leak occurs.
![memleak](https://cloud.githubusercontent.com/assets/7051550/12333319/d0d56414-baf3-11e5-8f12-f1b7ab0aca77.png)
Log - http://www1.zippyshare.com/v/RvC76mOy/file.html 11,5MB after unpacking.
Screens:
![1st logo](https://cloud.githubusercontent.com/assets/7051550/12333352/0173c278-baf4-11e5-852b-810917c96c4f.png)
![2nd logo](https://cloud.githubusercontent.com/assets/7051550/12333354/05b826a8-baf4-11e5-8d71-57b6540d5496.png)

**Comment by kobyaustin1** [comment]
has anyone figured out a fix to this yet?

**Comment by kobyaustin1** [comment]
mine plays music that is past rare logo but it freezes my computer so it didnt load

**Comment by CatoNator** [comment]
https://www.youtube.com/watch?v=0xlfJzZMZas

As of now, the memory leak no longer occures, but my CPU is maxed out and the performance is killed. The game makes some sounds after freezing, indicating that it might be still running and not completely hung.

**Comment by DrChat** [comment]
Yeah - it gets in-game fine. Just give it about an hour to compile shaders (not kidding!)

**Comment by SakataGintokiYT** [comment]
Do you already have compiled shaders ?

**Comment by CatoNator** [comment]
Are precompiled shaders even possible? I don't think they are.

**Comment by CatoNator** [comment]
The game's been compiling the shaders for about 1 hour and 20 minutes now, and it's in fact still doing it. So if I leave it alone for about two forevers tomorrow, and it gets the compiling done, do I have to compile the shaders every time I boot the game up or can I just skip that part somehow?

**Comment by xXHardGriefXx** [comment]
really want to play this game haha. its been compiling already for 15 minutes on a i7 3770. idk how long it will take haha. It has a blackish screen but my cpu is under 80-90% load.

**Comment by CatoNator** [comment]
https://www.youtube.com/watch?v=GOxMu1uNj5U

It can get into the menus now, but no further than that. Crashes after a little while no matter what you do.

**Comment by Atvriders** [comment]
i tested the game out today with the 3/7/17 build and it only gets past the rare logo and crashes at a gray screen

**Comment by BSoD38** [comment]
As of the 13/03/17 build (master/3cae25f) the game runs through the videos smoothly, with correct audio, but past those, I get many assertion errors :
![pic1](https://puu.sh/uHbt2/2f569ccbd9.png)

![pic2](https://puu.sh/uHbHG/a9148f11e4.png)

(Those two errors keep repeating themselves when I click ignore)

After clicking ignore many, many times, the errors stop for a while and the emulator is stuck on this screen :
![pic2](https://puu.sh/uHbUl/aa25f18e92.png)

Then I got errors again after about 5 minutes of waiting, and decided to close the emulator.

**Comment by MiniGrief** [comment]
Intro cut scenes play fine and if I can hear the main menu music with some glitchy video of spiral mountain. If I press start, it works but text is a broken mess and Banjo's room is also very buggy with some models loading fine, but textures being broken. Crashes soon after.

[https://www.youtube.com/watch?v=ZD-5zyl4OlE](url)

**Comment by jfarre20** [comment]
I was able to get in game by spamming A at the menu

![image](https://user-images.githubusercontent.com/7387914/29489967-f80032d4-84fb-11e7-8160-ebdfbd460244.png)

Runs at about 0.2fps and there isn't any visible 3d - but its progress :)

**Comment by sentriegun16** [comment]
after a few times of starting and crashing, i actually was able to see the intro textured correctly and was able to listen to the music, but it's slow and crashes soon after

**Comment by DrChat** [comment]
Some textures are being uploaded incorrectly. Possibly not handling mips correctly?

![renderdocui_2018-02-19_18-46-55](https://user-images.githubusercontent.com/2402369/36402593-49452ed0-15a5-11e8-9193-1fcb59314989.png)

![renderdocui_2018-02-19_18-48-19](https://user-images.githubusercontent.com/2402369/36402614-7ce62b18-15a5-11e8-9359-c1d41b4f78c9.png)

**Comment by DrChat** [comment]
Loading screen emulation isn't accurate enough.

# Xenia
![nab](https://user-images.githubusercontent.com/2402369/36462302-3835443e-1688-11e8-820c-0bcb2fc2345b.png)

# Xbox 360
![download 1](https://user-images.githubusercontent.com/2402369/36462308-449ab6e6-1688-11e8-8c69-a906212d2dc6.png)

**Comment by DrChat** [comment]
Aaand fixed the loading screen. 2_10_10_10 is actually 10_10_10_2.
![a](https://user-images.githubusercontent.com/2402369/36572028-2fcf46f0-17ff-11e8-9db6-72a7bfa58b87.png)

**Comment by Atvriders** [comment]
![image](https://user-images.githubusercontent.com/26294678/36884289-9955500a-1da5-11e8-88f5-2d9e72f4871a.png)

I got this far. my xbox one controller doesn't seem to work to push A.

**Comment by hubslave** [comment]
What graphics card do you have Atvriders? On my radeon hd 7700 it crashes after intro videos.

**Comment by Atvriders** [comment]
I have the GTX 1080

**Comment by hubslave** [comment]
Please try to spam start (it’s mapped to X button on keyboard) as soon as the rare logo starts. Someone was able to get to ingame menu by this.
PS: I attached debugger to xenia and crash on my computer is caused by invalid or somehow corrupted shader if it helps. Sometimes i can see that grey screen like BSoD38 behind jiggies.

**Comment by hubslave** [comment]
It's somehow working in benvanik/xenia@0cff9169. Some textures are correctly placed, but it runs for ~2 fps   
and does not appear to display any text or image. I am trying to run it again now, so i can document it and take some screenshots.

**Comment by hubslave** [comment]
IT'S HAPPENING
Sorry image wasn't captured, but it somehow rendered main menu 3d bg.
And then my computer crashed on VIDEO_DXGKRNL_FATAL_ERROR.

**Comment by Atvriders** [comment]
Does it really work now!

**Comment by hubslave** [comment]
Yes, but my radeon 7700 isn't the best, which is probably cause of those BSODs (i was able to create new save and it played intro sequence for 5 seconds and then BSOD)

**Comment by DrChat** [comment]
## Fonts
![b](https://user-images.githubusercontent.com/2402369/39607653-be299b92-4f01-11e8-9b6a-401ada05ea99.png)

### Font (originally 1920x1920)
![a](https://user-images.githubusercontent.com/2402369/39607693-09a774b8-4f02-11e8-9a0b-1c20576cc408.jpg)

<details><summary>Pixel Shader</summary><p>

```
/*    0.0 */       exec 
/*    9   */          add r0.___w, c1.xxxx, c1.xxxx
              +       maxs r1.__z_, r0.xx
/*   10   */          add r6.x___, r0.wwww, r0.xxxx
              +       maxs r2._, -r0.ww
/*   11   */          add r5._y__, r0.wwww, r0.yyyy
              +       adds_prev r0.__z_, r0.y
/*   12   */          add r3.x_z_, -r0.wwww, r0.xyyy
              +       maxs r5.___w, r5.yy
/*   13   */          max r5.x___, r3.xxxx, r3.xxxx
              +       maxs r5.__z_, r6.xx
/*   14   */          dp3 r7.x___, c3.zyxx, c255.yyyy
              +       maxs r3._y__, r0.yy
/*    0.1 */       exec 
/*   15   */          max r6._yz_, r0.yyzz, r0.yyzz
              +       maxs r1.___w, r5.yy
/*   16   */          tfetch2D r4.xyz_, r1.xy, tf1
/*   17   */          tfetch2D r8, r6.xy, tf0
/*   18   */          tfetch2D r9, r5.zw, tf0
/*   19   */          tfetch2D r10, r6.xz, tf0
/*   20   */          tfetch2D r11, r3.xy, tf0
/*    1.0 */       exec 
/*   21   */          tfetch2D r12, r5.xy, tf0
/*   22   */          tfetch2D r3, r3.xz, tf0
/*   23   */          tfetch2D r13, r0.xy, tf0
/*   24   */          tfetch2D r14, r1.zw, tf0
/*   25   */          tfetch2D r1, r0.xz, tf0
/*   26   */          serialize
                      sgt r0.___w, c1.zzzz, c255.xxxx
/*    1.1 */       exec 
/*   27   */          dp3 r5.x___, c4.zyxx, c255.yyyy
/*   28   */          dp3 r6.x___, c2.zyxx, c255.yyyy
/*   29   */          dp4 r1.x___, r1.wzxy, c0.wzxy
/*   30   */          dp4 r1._y__, r14.wzxy, c0.wzxy
/*   31   */          dp4 r1.__z_, r13.wzxy, c0.wzxy
/*   32   */          dp4 r3.x___, r3.wzxy, c0.wzxy
/*    2.0 */       exec 
/*   33   */          dp4 r3._y__, r12.wzxy, c0.wzxy
/*   34   */          dp4 r3.__z_, r11.wzxy, c0.wzxy
/*   35   */          dp4 r0._y__, r10.wzxy, c0.wzxy
/*   36   */          dp4 r0.x___, r9.wzxy, c0.wzxy
/*   37   */          dp4 r0.__z_, r8.wzxy, c0.wzxy
/*   38   */          add r4.xyz_, r4.zyxx, -r2.zyxx
/*    2.1 */       exec 
/*   39   */          mad r4.xyz_, r4.zyxx, c1.yyyy, r2.xyzz
/*   40   */          mul r7._yzw, r3.xxyz, c2.xxzy
              +       mulsc r5._y__, c4.x, r0.y
/*   41   */          mul r6._yzw, r1.xxyz, c3.xxzy
              +       mulsc r5.__z_, c4.z, r0.x
/*   42   */          add r6, r7.xzyw, r6.xzyw
              +       mulsc r5.___w, c4.y, r0.z
/*   43   */          add r5, r6, r5.xzyw
/*   44   */          dp3 r2.__z_, r5.yzww, c255.yyyy
              +       rcp r2.x___, r5.x
/*    3.0 */       exec   // PredicateClean=false
/*   45   */          mul r2.__z_, r2.zzzz, r2.xxxx
/*   46   */          max_sat r2.x___, r2.zzzz, r2.zzzz
              +       setp_ne r2._, r0.w
/*    3.1 */ (!p0) jmp L10
/*    4.0 */       exec   // PredicateClean=false
/*   47   */          sgt r0.x___, r3.xxxx, r1.zzzz
/*   48   */          setp_ne r2._, r0.x
/*   49   */     (p0) mad r2.__z_, r3.xxxx, c5.wwww, r2.xxxx
/*   50   */     (p0) subsc r0.x___, c255.y, r2.x
/*   51   */     (p0) mul r0.xyz_, r0.xxxx, c5.xyzz
/*   52   */     (p0) mad r4.xyz_, r2.xxxx, r4.xyzz, r0.xyzz
/*    4.1 */       jmp L15
                label L10
/*    5.0 */       exec   // PredicateClean=false
/*   53   */          sgt r0.___w, c1.wwww, c255.xxxx
/*   54   */          setp_ne r2._, r0.w
/*    5.1 */ (!p0) jmp L15
/*    6.0 */       exec 
/*   55   */          max r2._y__, r1.xxxx, r3.xxxx
/*   56   */          max r2._y__, r0.yyyy, r2.yyyy
/*   57   */          max r2._y__, r3.zzzz, r2.yyyy
/*   58   */          max r2._y__, r0.zzzz, r2.yyyy
/*   59   */          max r2._y__, r3.yyyy, r2.yyyy
/*   60   */          max r2._y__, r1.yyyy, r2.yyyy
/*    6.1 */       exec   // PredicateClean=false
/*   61   */          max r2._y__, r0.xxxx, r2.yyyy
/*   62   */          sgt r0.x___, r2.yyyy, r1.zzzz
/*   63   */          setp_ne r2._, r0.x
/*   64   */     (p0) mad r2.__z_, r2.yyyy, c5.wwww, r2.xxxx
/*   65   */     (p0) subsc r0.x___, c255.y, r2.x
/*   66   */     (p0) mul r0.xyz_, r0.xxxx, c5.xyzz
/*    7.0 */  (p0) exec 
/*   67   */     (p0) mad r4.xyz_, r2.xxxx, r4.xyzz, r0.xyzz
                label L15
/*    7.1 */       alloc colors
/*    8.0 */       exece 
/*   68   */          mul r4.___w, r2.zzzz, r2.wwww
/*   69   */          max oC0, r4, r4
/*    8.1 */       cnop
```

</p></details>

**Comment by CatoNator** [comment]
The game throws a "graphics device lost" error on me in the loading screen before the title screen loads. The build is 9a158963.

The log file reveals that the TextureCache staging buffer is too small.
[xenia.log](https://github.com/xenia-project/game-compatibility/files/1977622/xenia.log)

**Comment by xNano01x** [comment]
![bk 12](https://user-images.githubusercontent.com/39605085/40517543-3cebe8dc-5f8c-11e8-85b1-1352fde09a2d.PNG)
![terror](https://user-images.githubusercontent.com/39605085/40517663-e0b8c0fc-5f8c-11e8-9adc-ea3af6ac3c97.PNG)
![wow](https://user-images.githubusercontent.com/39605085/40517665-e0f39650-5f8c-11e8-9f78-1c124cd4615c.PNG)
![bk 10](https://user-images.githubusercontent.com/39605085/40517669-e1542e5c-5f8c-11e8-935c-daa0e1cab958.PNG)
![bk 1](https://user-images.githubusercontent.com/39605085/40517760-5ac30b5a-5f8d-11e8-870d-ea4fdc4e474c.PNG)

The game is being made playable, there are texture errors to review yet!
Water for example, and in some worlds, appear either white or yellow
The same is already a breakthrough for this game!

PC AMD: FX 8350 and R9 270X
:D

**Comment by CatoNator** [comment]
I can't manage to get beyond the title screen, as the game crashes due to the loss of the graphics device.

How on earth did you manage to get to actual gameplay though? The game runs at about 1 frame every 10 seconds for me. I'm using an i5 3570k and a GTX 970.

**Comment by xNano01x** [comment]
Patience and an xbox 360 joystick hahaha
Ami also goes to 1 FPS
@CatoNator

**Comment by xNano01x** [comment]
Gameplay!

https://www.youtube.com/watch?v=nQJvvv8vJ6Q

**Comment by Spokehedz** [comment]
I have gameplay, and some graphics corruption.

**Comment by some-username-here1** [comment]
Build: commit 37e59464 (commit aee5f582 with 0xFFFFFFFF in **xam_content.cc**)

# Issues:
audio lagging behind, and upon the jiggies loading screen, Xenia gives out a Internal GPU Lost error, but if left as is, without clicking on "OK", the music would start playing, but after 3 seconds or so, it would immediately cut off, but audio would still be playing, most likely from the vehicles in the titlescreen

Unknown if the Internal GPU Lost error is with just with Nvidia GPU's, as @Spokehedz was able to get in-game, or just incompatible recent drivers

# Logs:
[BK-N&B.log](https://github.com/xenia-project/game-compatibility/files/2482292/BK-N.B.log)

# Screenshots:
![BK-N&B](https://user-images.githubusercontent.com/34193156/47006025-db4cc900-d0e9-11e8-9be0-e3d8de27dbaf.png)

# Labels:
- (state-menus)

**Comment by some-username-here1** [comment]
**Build:** commit 00adfcad (Oct. 23, 2018, commit w/ 0xFFFFFFFF in **xam_content.cc**)

Downgraded the nvidia drivers down to Version 390.77 from 399.24+ and got Xenia working!

![Intro](https://user-images.githubusercontent.com/34193156/47384372-cceb4800-d6bb-11e8-9dcc-ffb645bc763b.png)

**Comment by some-username-here1** [comment]
Got into starting the story mode, but the screen has froze during the loading screen right after pressing A

![Frozen](https://user-images.githubusercontent.com/34193156/47387867-7551da00-d6c5-11e8-8ba5-dbadef53b8ca.png)

During the intro cutscene, the audio is normal but later the audio and music becomes desynced

**Comment by Brokemono** [comment]
![image](https://user-images.githubusercontent.com/46820617/51396750-fcd1ac80-1b3f-11e9-8622-c3692eb5bf84.png)
![image](https://user-images.githubusercontent.com/46820617/51396903-54701800-1b40-11e9-8883-1c1e52950698.png)
![image](https://user-images.githubusercontent.com/46820617/51396988-88e3d400-1b40-11e9-9944-9607ccc6ff69.png)
![image](https://user-images.githubusercontent.com/46820617/51397959-b0d43700-1b42-11e9-88f7-579aff23df66.png)
Lots of lag and stutter, grass didn't render right and some other stuff like those decorations on the buildings and some vines, 30 minutes of gameplay and no crashes to report. This is running on the DX12 version and the latest build as of 1/17/2019. Despite all of this I made it to the highest place in this world. There is also slight input delay but that's to be expected at this framerate and stutter. It averages around 13fps most of the time. and sometimes going even above 15fps but after it hits 15+ fps after a second or so it hangs for ca. 3 seconds and then it goes to normal.

Specs:
CPU: AMD FX-8300 (overclocked to 3.7Ghz)
GPU: Palit Nvidia GeForce 670
RAM: GSkill Ripjaws X Series 2 x 8Gb Dual Channel
Motherboard: GA-78LMT-USB3
OS: Windows 10 Pro Version 1803 (OS Build 17134.523)

PS: I had Chrome, Discord, Steam and MSI Afterburner running in the background while testing the game.

**Comment by Atvriders** [comment]
Thanks. I'll have to get that version. The game runs for me but audio lags and extreme texture glitching.

**Comment by Brokemono** [comment]
![image](https://user-images.githubusercontent.com/46820617/51609262-f1083080-1f19-11e9-96b8-b45663a10dda.png)
![image](https://user-images.githubusercontent.com/46820617/51609300-0b420e80-1f1a-11e9-966b-95728e0f9099.png)
![image](https://user-images.githubusercontent.com/46820617/51609371-43495180-1f1a-11e9-963e-2062b95ca8f2.png)
![image](https://user-images.githubusercontent.com/46820617/51609649-ebf7b100-1f1a-11e9-8595-34ecd4f23521.png)
![image](https://user-images.githubusercontent.com/46820617/51609828-69232600-1f1b-11e9-8bdd-0a112639bba0.png)
![image](https://user-images.githubusercontent.com/46820617/51609871-7dffb980-1f1b-11e9-9710-9ec683046fd4.png)
![image](https://user-images.githubusercontent.com/46820617/51609896-8e179900-1f1b-11e9-85f7-bac961919fea.png)
Compared to my last one there is quite some improvement here. No more texture issues but the grass still isn't rendered right but it's not that big of a problem right now. The fps averaged 13 this time but it had far fewer stutters and felt smoother but there was still this audio delay and input delay. but I climbed the tower faster. This time I ran xenia at high priority and at full-screen. My monitor is 1680x1050 which is 16:10 Aspect Ratio...The game looks fine with it. I think someone with better specs than mine and better drivers and all that could actually play this smoothly right now.

Specs:
CPU: AMD FX-8300 (overclocked to 3.7Ghz)
GPU: Palit Nvidia GeForce 670
RAM: GSkill Ripjaws X Series 2 x 8Gb Dual Channel
Motherboard: GA-78LMT-USB3
OS: Windows 10 Pro Version 1803 (OS Build 17134.523)

PS: I yet again had Steam, Discord, Chrome and MSI Afterburner running in the background.

**Comment by thareh** [comment]
The grass goes where it's not supposed to go, the skybox in the first level doesn't show at all and some UI stuff that doesn't show sometimes but otherwise it works pretty damn well! Getting almost max FPS on my computer, can't wait for it to work fully - this is my favorite game!

**Comment by some-username-here1** [comment]
![screenshot 95](https://user-images.githubusercontent.com/34193156/53380027-c9452780-3920-11e9-82c1-46ef1cb24769.png)
![screenshot 96](https://user-images.githubusercontent.com/34193156/53380028-c9ddbe00-3920-11e9-8e0e-b3170268c2fc.png)
![screenshot 97](https://user-images.githubusercontent.com/34193156/53380030-c9ddbe00-3920-11e9-83fa-ae90bf6b3f3f.png)
![screenshot 98](https://user-images.githubusercontent.com/34193156/53380031-c9ddbe00-3920-11e9-95bb-bc55ce8d5080.png)
![screenshot 99](https://user-images.githubusercontent.com/34193156/53380032-c9ddbe00-3920-11e9-9705-f8baca48f53d.png)
![screenshot 100](https://user-images.githubusercontent.com/34193156/53380034-ca765480-3920-11e9-9395-5df5c339c1c4.png)
![screenshot 101](https://user-images.githubusercontent.com/34193156/53380035-ca765480-3920-11e9-96b6-836dbf581e54.png)
![screenshot 102](https://user-images.githubusercontent.com/34193156/53380037-cb0eeb00-3920-11e9-8a23-8711bfcca37a.png)
![screenshot 103](https://user-images.githubusercontent.com/34193156/53380038-cba78180-3920-11e9-9a9a-2b4e5445e5f9.png)
![screenshot 104](https://user-images.githubusercontent.com/34193156/53380039-cba78180-3920-11e9-98f7-105cac377251.png)
![screenshot 105](https://user-images.githubusercontent.com/34193156/53380272-ab2bf700-3921-11e9-8790-d971fd189e5d.png)
![screenshot 106](https://user-images.githubusercontent.com/34193156/53380273-ab2bf700-3921-11e9-80db-e99a0902a8bd.png)
![screenshot 107](https://user-images.githubusercontent.com/34193156/53380274-ab2bf700-3921-11e9-9eff-3d5be7cadd95.png)

# Issues:
When starting out fromt the intro, the sound effects, music and voicelines are all off. But the music is ahead of both, while the sound effects are slightly behind (or ahead, don't remember) of the voicelines and cutscene.

Once we're travelled to showdown town, though, the game crashes.

# Specs:
* OS: Windows 8.1 Professional (64-bit)
* CPU: Intel Core i5-4590 (3.70 GHz)
* RAM: DDR3-1866 16GB (2 x 8GB)
* GPU: AMD Radeon R7 250 (via DVI to HDMI)
* * Driver Version: 19.1.1 (yes, the Windows 7 drivers still works with 8.1 for whatever reason)

# Log:
[BK-N&B.zip](https://github.com/xenia-project/game-compatibility/files/2903346/BK-N.B.zip)

**Comment by Richard64x** [comment]
Issues:

The audio is desynced. There are no obvious graphical issues. Unfortunately, at the start of the first loading screen at the wall of jiggies, the game crashes, showing a "graphics device lost" error


Specs:

    OS: Windows 10 Education (64-bit)
    CPU: Intel Core i5 3340 @ 3.10GHz
    RAM: 8GB DDR3
    GPU: GTX 1050
[xenia.log](https://github.com/xenia-project/game-compatibility/files/3366063/xenia.log)


Log:

**Comment by Richard64x** [comment]
Tested on: https://github.com/xenia-project/xenia/commit/3238fae75849aac0637d61b087994177dd8ca5a8

Issues:
Crashes at the beginning of the first loading screen, right after the Rare logo.
Audio is desynced by about a second or so
No obvious graphical issues

Log:

Labels:
state-intro
regression

**Comment by Crazedtech** [comment]
So i tried this for a little today because i can't find the power cable for my xbox 360, here's my results
**Issues:**
Grass is off position, something that's been said here before
on the way up the tower i could see to the left on the ledge at the foot of the ramp was alot of sugarcane-looking glitched grass
Mild input delay (1-0.5 secs)
Horrible audio delay (sometimes various sound effects just don't play, others take about a minute to play unless cancelled

Tested on DirectX12 version of Xenia
I'd supply a log, but i re-opened the game after this session, erasing the old one i assume

Specs:
Palit GTX 1050ti StormX (No extra overclocking beside Palit mod)
Ryzen 3 1200
No sound cards installed, running off Asus A320m-k mobo built-in sound

**Comment by Crazedtech** [comment]
Another round of testing, but this time i tried to do the first Jiggy challenge. Performance is really bad in the first level, and graphics device was lost after completing the challenge and getting to the farm.

[BanjoLog.zip](https://github.com/xenia-project/game-compatibility/files/3372575/BanjoLog.zip)

**Comment by momermop** [comment]
Hello tried to test the game preformance was a bit thuggy laggy the loading screens were a bit long sometimes could play the game till the logbox world after the mungo antenna mission it crashed 

(the log file is 2,75 GB so i can't upload it but i can upload the last part with the crash dump
[Xenia (small part including the crash dump).txt](https://github.com/xenia-project/game-compatibility/files/3779921/Xenia.small.part.including.the.crash.dump.txt)

GTX 2070 Super
I5-8600k
16GB DDR4 Ram

if anyone wants the full log here is it on google drive
https://drive.google.com/file/d/1HoyFps7qtCSjKn34i2ca9dkdvGYhSHF6/view?usp=sharing
it is 2,8 GB so ALOT of text :)

Note this log is from more then 1,5 hour gameplay without 1 crash so thats why its pretty big

**Comment by Margen67** [comment]
@momermop You can put the log in a zip to make it smaller.

**Comment by momermop** [comment]
A yeah mb i uploaded it now to google drive since it is 40 MB and still to large for github can't compress it any smaller

https://drive.google.com/open?id=1DCHsHGw2gdZXRmnR1bi6s1z3Xc5xBB_Y

**Comment by raulib** [comment]
Intro works fine as fuck, but after rare logo and sounds and music with trumpets the game just crashes.

**Comment by momermop** [comment]
i dunno i could play to the second world but after that is crashed.

the game worked ok but the sound had a 7- 8 second delay

**Comment by raulib** [comment]
Very stable, but has lots of throttling and lag, and because shaders aren't preloaded that might be the culprit. Also the witch car is your enemy from the 1st minute

![](https://cdn.discordapp.com/attachments/309008096800276490/686304992470368299/unknown.png)

![](https://cdn.discordapp.com/attachments/309008096800276490/686305016679759873/unknown.png)

![](https://cdn.discordapp.com/attachments/309008096800276490/686305044613693449/unknown.png)

![](https://cdn.discordapp.com/attachments/309008096800276490/686305077144977553/unknown.png)



And the car spinning out



![](https://cdn.discordapp.com/attachments/309008096800276490/686305099861327915/unknown.png)

Also, the icon for wheels, engine, fuel and cart stutter


![](https://cdn.discordapp.com/attachments/309008096800276490/686306570275127341/unknown.png)

![](https://cdn.discordapp.com/attachments/309008096800276490/686306432458817595/unknown.png)

![](https://cdn.discordapp.com/attachments/309008096800276490/686306412313444437/unknown.png)

![](https://cdn.discordapp.com/attachments/309008096800276490/686306355824295956/unknown.png)


So fat, it works good, closed game after building cause of lag and because I got to go.

**Comment by nicklas321123** [comment]
anybody know why i just get a white screen after this shows up? seems to be as far as i am able to get
![image](https://user-images.githubusercontent.com/44222195/78486835-ec89fb00-770c-11ea-9a24-e8a5027e3159.png)

**Comment by General-lroh** [comment]
I've got the game working fine and smooth with some minor graphical errors but its not really an issue. But when I try to save a vehicle blueprint the game just brings up a save tab that I can't click on or rename and it doesn't go away even when I leave the garage. Any ideas on how to get past this?
![BKNB](https://user-images.githubusercontent.com/64738912/80916613-b19dd600-8d51-11ea-9581-3699a92425c7.png)

**Comment by dementedchiken** [comment]
Hi Guys, 

I've just found this thread and see it's still somewhat active. I'm very new to all this, but wondered if you had any help you could offer. I'm running a 5700xt and a Ryzen 3600x with the latest drivers and 16gb or ram. However no matter which graphical API I use I get weird CTDs and strange bugs.

Any help would be very apreciated and apologies if I'm in the wrong place

**Comment by Margen67** [comment]
@dementedchiken Tech support belongs on the Discord server: https://discord.gg/Q9mxZf9

**Comment by NorliaTiemeth** [comment]
So here's what I found out, Xenia doesn't like Windows 7 when running this game, however with Windows 10 all of the graphical bugs are solved, however I still get some bad framerate issues but for the most part the game is running fine. Does anyone know how I could get the frame stutter to stop all together? Specs listed below.
SPECS:
I5-4690K running at 4.5Ghz
Sapphire Nitro RX470 running at 1306mhz core and 1750mhz memory
16GB (4x4) DDR3 Ram running at 1600mhz
Kingston A400 120gb SSD Sata Gen 3
WD Blue 1TB 7.2k rpm HDD Sata Gen 3
2x Seagate Barracuda 7.2k rpm 500GB HDD Sata Gen 2

**Comment by FoxHunter267** [comment]
How can i get the lastest version of this game?

**Comment by Stryxus** [comment]
@FoxHunter267 legally.

**Comment by Tylertron1998** [comment]
anyone have any good configs for this game specifically? Tried grabbing latest master build and ran today, very laggy, stuttery, and hard to play. Running on a Ryzen 5 3600, RTX 2060S, win10.

**Comment by andrewspalato** [comment]
@Tylertron1998 You should use Xenia Canary, performance is a bit better, getting 21 or so fps on a GTX 1060. You might as well set the DX12 resolution scale to 2 in the config as well :)

**Comment by Tylertron1998** [comment]
@andrewspalato unfortunately that hasn't helped much - getting around 10fps at best.

**Comment by Spokehedz** [comment]
I get better than 10FPS on my RX580 and FX8130... The opening sequence was 30fps for most of it, and Showdown Town is 18-22ish.

**Comment by Tylertron1998** [comment]
Interesting, not sure where my issue is - happy to provide reports to anyone interested.

**Comment by mARvOEoUS** [comment]
Getting for average 50 fps and near 60 with vsync off staring at a wall then staring at everthing else
![nb](https://user-images.githubusercontent.com/36290872/102709228-490f6380-42f8-11eb-8f09-2768b42cbaa3.PNG)

![bn](https://user-images.githubusercontent.com/36290872/102709233-5298cb80-42f8-11eb-8ab1-a079f0f470ec.PNG)

i7-8700K running at 3.7Ghz
EVGA FTW3 1080ti at 1961mhz GPU Clock and 5508mhz Memory Clock
32GB (4x4) DDR4 Ram running at 2400mhz

**Comment by andrewspalato** [comment]
@Tylertron1998 I actually have Banjo Kazooie Nuts and Bolts title update 3 installed with xenia, which may help with performance just a little.

**Comment by mARvOEoUS** [comment]
> @Tylertron1998 I actually have Banjo Kazooie Nuts and Bolts title update 3 installed with xenia, which may help with performance just a little.

How does one get the update

**Comment by andrewspalato** [comment]
@mARvOEoUS This works: You have to source the title updates yourself. Google can help you out though ;)
[https://youtu.be/6EyaYWuml-c?t=301](url)

**Comment by mARvOEoUS** [comment]
@andrewspalato 
Awesome, the update gave a tiny performance boost. Staring at wall now is average 60fps and 74fps average if zoomed into Banjo and before it was 50fps. Staring at "everything else" now average 23fps but less stuttering. Good stuff

**Comment by mARvOEoUS** [comment]
The game starts to corrupt while after some playtime: textures start to become "missing no", models start spazzing and rarely cause crashes

**Comment by andrewspalato** [comment]
Me too, some parts like propellers/blocks etc on vehicles become partly corrupted and glitchy when leaving the training ground with a loaded vehicle. Loading the vehicle from inside a world doesn't cause these glitches. UI button sprites can glitch out from there also.

**Comment by Ch1pless** [comment]
On running the first time, experiences extreme stuttering issues. From looking at the log, appears to be related to loading textures. Warnings that repeated in multiple locations to assist: 
 - F8000004 GPU: Write to unknown register (At beginning of launch)
 - F8000004 Texture fetch constant has "invalid" type! (Through most of the log)
The game was ran from bootup to menu screen.
[xenia_log.zip](https://github.com/xenia-project/game-compatibility/files/6241750/xenia_log.zip)

**Comment by jarodhine** [comment]
As of xenia-project/xenia@1a228289 Banjo Kazooie Nuts & Bolts is fully playable to 100% completion excluding DLC and Stop N Swap with minimal crashes and errors on my system.

CPU: Stock Ryzen 3600
GPU: Nvidia 2060
Memory: 16 GB 3600 MHz
OS: Windows 10 Version 10.0.19042 Build 19042

Errors:
Fps dips to 20 facing busy sections of showdown town.
Vehicle part flickering and vertex minor vertex explosion occasionally when exiting Mumbo Motors, Fixed by reloading vehicle in world.
A total of 5 crashes took place during my 21 hour playtime.

Save Folder: 
[4D5307ED.zip](https://github.com/xenia-project/game-compatibility/files/6545473/4D5307ED.zip)


![BanjoKazooieNandB](https://user-images.githubusercontent.com/26563613/119636002-910e5580-be57-11eb-96c2-95a4df94bbc9.png)

**Comment by Spokehedz** [comment]
Thanks for that save, really nails home the "works 100%" aspect.

**Comment by Wizerdz** [comment]
Current status?

**Comment by xZaino** [comment]
Where do you put the save files at though? I have been having trouble with the save files. I ran my Xbox Save Files through Horizon but, that did not work. Have any ideas? I would like my vehicles and my save file on the emulator so I can do Vehicles Around Town.

**Comment by xZaino** [comment]
Okay, so I got Banjo Kazooie Nuts And Bolts working. I figured out what I had to do and it I launched the game. But, the game save didn’t work. Or the vehicles. When I pressed Y on the main menu screen to select a saved game, it told me  the game file was corrupt and I needed to overwrite the game files to play. I found out that the vehicles were corrupt when I put in a 100 % game save into the documents folder and launched the game to see the vehicles. Mumbo told me that the vehicles were Corrupt lol. I think this may be because I’m downloading it from the cloud drive into the flash drive. I don’t know if it will change if I download it locally on the Xbox 360 drive and maybe see if it would work. Any thoughts?

**Comment by ZolaKluke** [comment]
Has working splitscreen
![image](https://user-images.githubusercontent.com/6497775/128720437-2a3a7025-6852-4bc0-ac92-162c88d7558a.png)

### Labels:
splitscreen-working

**Comment by ChargingTurnip** [comment]
For some reason, this spot in Nutty Acres is not rendered, unlike on console. Not sure if this is actually something worthy enough to be posted here, but I just thought I'd give it a mention.
![image](https://user-images.githubusercontent.com/53988299/133912446-44e8eb52-cff3-4d90-a163-1c45d8161e8e.png)

**Comment by OneguyofOR** [comment]
Hi, could you please say why does the game stays on the loading screen?
![image](https://user-images.githubusercontent.com/90712257/156095929-115d37c1-ab82-4f42-99b8-fbe936247b73.png)

**Comment by GavinKeane** [comment]
The mission "The Saucer of Peril Returns" in the terrarium of terror plays fine until the very end. Once the saucer reaches the end of its path, the game hangs forever and the mission complete screen never appears. 
![scn](https://user-images.githubusercontent.com/32646750/227243813-f4aca62c-d3f4-446a-9d24-50f687abd8c1.png)

**Comment by TheAshmanCometh** [comment]
The game has no major visual bugs for me on Master, but Canary has garbled squares on a parked vehicle, the minimap, and all over the screen during vehicle construction. The ground texture in Nutty Acres also flickers rapidly.

**Comment by goopstack** [comment]
was able to fully play through the game on canary without too many major issues. occasional white square in mumbos motors, some HUD not showing up at times, and a couple crashes in my 8 hours of game time. Definitely playable with the patch. I had consistent 60fps with vsync on (few seconds of slowdown when entering a world for the first time, re-entering did not slow down after that) with resolution x and y set to 2. my build:
RTX 3060ti
i7-9700k
32 GB RAM

save file
[4D5307ED.zip](https://github.com/xenia-project/game-compatibility/files/11855797/4D5307ED.zip)

**Comment by samcarsonx** [comment]
Game normally runs almost perfectly. However, changing resolution x and y to 2 causes special effects (eg glow) to appear corrupt for me, and the characters' heads in dialogue boxes flicker. I understand that there is a warning regarding visual bugs with changing the resolution in the config file, but it seems other people have no issues so maybe it's worth noting.

My build: RTX 3070, i7 12700F, 32GB DDR4 RAM, 1TB NVMe SSD
Running Xenia [v1.0.2808](https://github.com/xenia-project/release-builds-windows/releases/tag/v1.0.2808-master)

Selected menu item:
![banjo1](https://github.com/xenia-project/game-compatibility/assets/37749448/41de8cf3-c00e-442c-ab0f-5e275bec459d)
Minimap:
![banjo2](https://github.com/xenia-project/game-compatibility/assets/37749448/a8a86191-de8a-4c99-bb27-6d1274419c94)
Nearby object that can be picked up:
![banjo5](https://github.com/xenia-project/game-compatibility/assets/37749448/959689f5-d9ce-404d-ab35-411476843cd6)

**Comment by OneguyofOR** [comment]
Great to see another Xbox exclusive game working!

-original message-
Subject: Re: [xenia-project/game-compatibility] 4D5307ED - Banjo Kazooie: Nuts & Bolts (#49)
From: Sam Carson ***@***.***>
Date: 08/02/2024 22:53

Game normally runs almost perfectly. However, changing resolution x and y to 2 causes special effects (eg glow) to appear corrupt for me, and the characters' heads in dialogue boxes flicker. I understand that there is a warning regarding visual bugs with changing the resolution in the config file, but it seems other people have no issues so maybe it's worth noting.

My build: RTX 3070, i7 12700F, 32GB DDR4 RAM, 1TB NVMe SSD
Running Xenia [v1.0.2808](https://github.com/xenia-project/release-builds-windows/releases/tag/v1.0.2808-master)

Selected menu item:
![banjo1](https://github.com/xenia-project/game-compatibility/assets/37749448/41de8cf3-c00e-442c-ab0f-5e275bec459d)
Minimap:
![banjo2](https://github.com/xenia-project/game-compatibility/assets/37749448/a8a86191-de8a-4c99-bb27-6d1274419c94)
Nearby object that can be picked up:
![banjo5](https://github.com/xenia-project/game-compatibility/assets/37749448/959689f5-d9ce-404d-ab35-411476843cd6)

**Comment by Enterprise12nx01** [comment]
> Game normally runs almost perfectly. However, changing resolution x and y to 2 causes special effects (eg glow) to appear corrupt for me, and the characters' heads in dialogue boxes flicker. I understand that there is a warning regarding visual bugs with changing the resolution in the config file, but it seems other people have no issues so maybe it's worth noting.
> 
> My build: RTX 3070, i7 12700F, 32GB DDR4 RAM, 1TB NVMe SSD Running Xenia [v1.0.2808](https://github.com/xenia-project/release-builds-windows/releases/tag/v1.0.2808-master)
> 
> Selected menu item: ![banjo1](https://private-user-images.githubusercontent.com/37749448/303411513-41de8cf3-c00e-442c-ab0f-5e275bec459d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxMDQyNDgsIm5iZiI6MTcxNjEwMzk0OCwicGF0aCI6Ii8zNzc0OTQ0OC8zMDM0MTE1MTMtNDFkZThjZjMtYzAwZS00NDJjLWFiMGYtNWUyNzViZWM0NTlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE5VDA3MzIyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5N2RkYmM1MmM0YmZjODM5NjdmNmMzNTExNWI3NDVlNjQzMmVkNzk5Mzk5YWNjNTE5YWZhZGQ5ZDEzYjVhNTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4hbxKq_nBprowvjptHtDCh50vQEvjfDDtka82BR-eQk) Minimap: ![banjo2](https://private-user-images.githubusercontent.com/37749448/303411511-a8a86191-de8a-4c99-bb27-6d1274419c94.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxMDQyNDgsIm5iZiI6MTcxNjEwMzk0OCwicGF0aCI6Ii8zNzc0OTQ0OC8zMDM0MTE1MTEtYThhODYxOTEtZGU4YS00Yzk5LWJiMjctNmQxMjc0NDE5Yzk0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE5VDA3MzIyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlOTExNTA4ZmM1YzFjZTYzODJlOWYzZDdmYjA4YjEyMWI5OTY4ZWY5NTA2YjE5MGQ1YmM4NGVmMjVmZmE5MWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.99l6AueFzdvYK2L_NAR5xI6LcPgKKb6Arnv0-ghiT60) Nearby object that can be picked up: ![banjo5](https://private-user-images.githubusercontent.com/37749448/303412505-959689f5-d9ce-404d-ab35-411476843cd6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxMDQyNDgsIm5iZiI6MTcxNjEwMzk0OCwicGF0aCI6Ii8zNzc0OTQ0OC8zMDM0MTI1MDUtOTU5Njg5ZjUtZDljZS00MDRkLWFiMzUtNDExNDc2ODQzY2Q2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE5VDA3MzIyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjNzRjY2VkNzFlZTI1ZmMzYTk4MDc3NDE2MTMzNGZjOWU3NjI2MjFmOTNiMWZiZDFkM2NlOGMyNGE1MmQ0OTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3OSs3_DK6A7dyDkuPxGl8h2O16bQOVti_S64oJo5A9c)

i also had this issue currently, lots of stutter 5800x3d and 7900 xtx

**Comment by TheDuckKing** [comment]
running on latest build of xenia canary currently available. (released May 30th 2024) pc specs: Ryzen 5 7600x. 32GB DDR5 5600MHZ ram, RTX 4070ti
when closing out of xenia and coming back to the game all collectibles are wiped from save otherwise runs fine at native resolution with little to no graphical bugs

**Comment by Atvriders** [comment]
![17221490019314958393401010018133](https://github.com/user-attachments/assets/cf834f4f-1774-47f7-aeb0-c7eb89ee149a)
Running around 30 FPS on steam deck OLED. Engine parts sometimes glitch. Otherwise game is running pretty good. The tutorial is very laggy. Once you get past it runs fine.

**Comment by russandro** [comment]
[samcarsonx](https://github.com/samcarsonx) commented [on Feb 8](https://github.com/xenia-project/game-compatibility/issues/49#issuecomment-1934538487)
> Game normally runs almost perfectly. However, changing resolution x and y to 2 causes special effects (eg glow) to appear corrupt for me, and the characters' heads in dialogue boxes flicker.
Same here.

**Comment by russandro** [comment]
Is there maybe any patch you know that fixes this?

**Comment by Supercoder219** [comment]
as of now it seems to be able to run it almost flawlessly, other than some graphical glitches that can be fixed by clearing gpu runtime cache

**Comment by Arlena4000** [comment]
Game runs great for me too,  as long as you set up-scaling x and y to default 1.
Occasionally though I get weird graphical bugs but that can be sorted through restarting a world/mission. If it happens again I'll upload a screenshot.

However you can enable internal_display_resolution to = 16, which is 1080p. I don't know if this actually makes it render at 1080p but I think it does?
Also if you experience screen tearing whilst fullscreen enable this setting to false, vsync on
d3d12_allow_variable_refresh_rate_and_tearing = false

**Comment by Margen67** [comment]
@Arlena4000 Xenia Canary discussion belongs here: Xenia Canary discussion belongs here: https://github.com/xenia-canary/game-compatibility

**Comment by IsaacWild** [comment]
I have been running this on my SteamDeck and it starts and looks great running at 30FPS but it crashes on the first level after the game master says hi and I get this crash message. I'm very new to emulation is there anything I can do to fix this?

![Image](https://github.com/user-attachments/assets/d77d808c-182b-40d0-abe9-e189f5585af5)
